app/policies/test_result_policy.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Policies for accessing TestResults
+class TestResultPolicy < ApplicationPolicy
+  def index?
+    Pundit.policy(user, record.profile)
+  end
+
+  def show?
+    Pundit.policy(user, record.profile)
+  end
+
+  # Only show TestResults belonging to profiles owned by the current user
+  class Scope < ::ApplicationPolicy::Scope
+    def resolve
+      available_profiles = HostPolicy::Scope.new(
+        user, ::Profile
+      ).resolve.pluck(:id)
+      scope.where(profile_id: available_profiles)
+    end
```

Comment:
It's much better to include `available_profiles` as a sub-query than a list of IDs. The resulting query is of this form:

```sql
SELECT "test_results".* FROM "test_results" WHERE "test_results"."profile_id" IN (SELECT "profiles"."id" FROM "profiles" WHERE "profiles"."account_id" = '0cda8d0d-0ef5-4b00-92b5-17be60ed68b9')
```

```suggestion
    def resolve
      available_profiles = HostPolicy::Scope.new(
        user, ::Profile
      ).resolve
      scope.where(profile: available_profiles)
    end
```